tests/tests_integration_pytest.yml

```diff
@@ -0,0 +1,82 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Empty play to gather facts
+  hosts: all
+
+- import_playbook: playbooks/rhel7_deps.yml
+  when:
+    - ansible_distribution == "RedHat"
+    - ansible_distribution_major_version == '7'
+
+- import_playbook: playbooks/centos6_deps.yml
+  when:
+    - ansible_distribution == "CentOS"
+    - ansible_distribution_major_version == '6'
+
+- import_playbook: playbooks/centos7_deps.yml
+  when:
+    - ansible_distribution == "CentOS"
+    - ansible_distribution_major_version == '7'
+
+- import_playbook: playbooks/centos8_deps.yml
+  when:
+    - ansible_distribution == "CentOS"
+    - ansible_distribution_major_version == '8'
+
+- name: Run test playbook
+  hosts: all
+  vars:
+    - rundir: /run/system-roles-test
+  tasks:
```

Comment:
Instead of having separate playbooks for each platform, with the duplicated code, it would be better to have vars files for each platform defining the packages / services for each platform.  https://github.com/oasis-roles/meta_standards#platform-specific-variables - something like this:
* make a `vars/integration/` subdirectory of `tests/`
* create the files `vars/integration/CentOS-6.yml` etc. - one file for each supported platform

Each file would look something like this:
```yaml
__test_pre_packages: [epel-release]
__test_packages: [git, python3-pip, ansible]
__test_services: [rsyncd]
__test_pip_packages: [pytest]
```
of course these will be different depending on the platform - in addition, for epel support on rhel7, you may have to specify the url of the rpm directly e.g. `__test_pre_packages: ["https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm"]` as I don't think the package `epel-release` is in RHEL7.
Then, in `tests/tests_integration_pytest.yml` do something like this:
```yaml
- name: Set platform/version specific variables for testing
  include_vars: "{{ __test_vars_file }}"
  loop:
    - default.yml
    - "{{ ansible_facts['os_family'] }}.yml"
    - "{{ ansible_facts['distribution'] }}.yml"
    - "{{ ansible_facts['distribution'] }}_{{ ansible_facts['distribution_major_version'] }}.yml"
    - "{{ ansible_facts['distribution'] }}_{{ ansible_facts['distribution_version'] }}.yml"
  vars:
    __test_vars_file: "{{ role_path }}/tests/vars/integration/{{ item }}"
  when: __test_vars_file is file

- name: ensure test pre-packages are installed first
  package:
    name: "{{ __test_pre_packages }}"
  when: __test_pre_packages is defined

- name: ensure test packages are installed
  package:
    name: "{{ __test_packages }}"
  when: __test_packages is defined

- name: ensure test services are enabled
  service:
    name: "{{ item }}"
    state: started
    enabled: true
  loop: "{{ __test_services | d([]) }}"

- name: ensure pip packages are installed
  pip:
    name: "{{ __test_pip_packages }}"
  when: __test_pip_packages is defined
```

Reply:
It does look like a more clean way of stating everything, I'm going to try to change it this way, thanks!!